p2p/src/peer_manager/peerdb/address_data/mod.rs

```diff
@@ -152,7 +160,11 @@ impl AddressData {
             MAX_DELAY_REACHABLE
         };
 
-        std::cmp::min(fail_count * Duration::from_secs(60), max_delay)
+        // 10, 20, 40, 80... seconds
+        std::cmp::min(
+            2u32.saturating_pow(fail_count) * Duration::from_secs(10),
```

Comment:
This number can easily overflow... fail_count only has to be > 32 for that to happen, with or without multiplication... please guard the operations with whatever necessary to ensure this can't happen.

Reply:
It should not overflow because seconds in `Duration` are represented as u64. Replaced by `saturated_mul` anyway.